File: app/src/main/java/com/ilya4/rssreader/Entity.java
package com.ilya4.rssreader;

import android.support.annotation.NonNull;

import java.text.DateFormat;
import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class Entity implements Comparable<Entity>{

    private String image;
    private String title;
    private String link;
    private Date pubDate;



    public Entity(String image, String title, String link, String pubDate) {
        this.image = image;
        this.title = title;
        this.link = link;
        DateFormat format = new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss z", Locale.ENGLISH);
        Date date = new Date();
        try {
            date = format.parse(pubDate);
        }catch (ParseException e){
            e.printStackTrace();
        }
        this.pubDate = date;
    }

    public String getImage() {
        return image;
    }

    public String getTitle() {
        return title;
    }

    public String getLink() {
        return link;
    }


    @Override
    public int compareTo(@NonNull Entity o) {
        return pubDate.compareTo(o.pubDate);
    }
}

File: app/src/main/java/com/ilya4/rssreader/Entity.kt
package com.ilya4.rssreader


import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class Entity(val image: String, val title: String, val link: String, pubDate: String) : Comparable<Entity> {
    private val pubDate: Date


    init {
        val format = SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss z", Locale.ENGLISH)
        var date = Date()
        try {
            date = format.parse(pubDate)
        } catch (e: ParseException) {
            e.printStackTrace()
        }

        this.pubDate = date
    }


    override fun compareTo(o: Entity): Int {
        return pubDate.compareTo(o.pubDate)
    }
}
